### ISSUE TITLE
timeout at starting a call

### ISSUE THREAD

**Issue by endredy** [asterisk-java/asterisk-java#302]
Hi,
there is a bug (with the latest version), which sometimes occurs in our log at heavy traffic (more than 5-10 calls start at the same millisec).

When a new channel is added, a timeout happens:

```
08:05:20,709 ERROR [stderr] (AJ DaemonPool-1.2) Exception in thread "AJ DaemonPool-1.2" org.asteriskjava.live.ManagerCommunicationException: Unable to send GetVarAction
08:05:20,709 ERROR [stderr] (AJ DaemonPool-1.2) 	at org.asteriskjava.live.internal.ManagerCommunicationExceptionMapper.mapSendActionException(ManagerCommunicationExceptionMapper.java:60)
08:05:20,709 ERROR [stderr] (AJ DaemonPool-1.2) 	at org.asteriskjava.live.internal.AsteriskServerImpl.sendAction(AsteriskServerImpl.java:985)
08:05:20,709 ERROR [stderr] (AJ DaemonPool-1.2) 	at org.asteriskjava.live.internal.AsteriskChannelImpl.getVariable(AsteriskChannelImpl.java:763)
08:05:20,709 ERROR [stderr] (AJ DaemonPool-1.2) 	at org.asteriskjava.live.internal.ChannelManager.addNewChannel(ChannelManager.java:223)
08:05:20,709 ERROR [stderr] (AJ DaemonPool-1.2) 	at org.asteriskjava.live.internal.ChannelManager.handleNewChannelEvent(ChannelManager.java:496)
08:05:20,709 ERROR [stderr] (AJ DaemonPool-1.2) 	at org.asteriskjava.live.internal.AsteriskServerImpl.onManagerEvent(AsteriskServerImpl.java:1045)
08:05:20,709 ERROR [stderr] (AJ DaemonPool-1.2) 	at org.asteriskjava.manager.ManagerEventListenerProxy$1.run(ManagerEventListenerProxy.java:59)
08:05:20,709 ERROR [stderr] (AJ DaemonPool-1.2) 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
08:05:20,709 ERROR [stderr] (AJ DaemonPool-1.2) 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
08:05:20,709 ERROR [stderr] (AJ DaemonPool-1.2) 	at java.lang.Thread.run(Thread.java:745)
08:05:20,725 ERROR [stderr] (AJ DaemonPool-1.2) Caused by: org.asteriskjava.manager.TimeoutException: Timeout waiting for response to GetVar
08:05:22,033 ERROR [stderr] (AJ DaemonPool-1.2) 	at org.asteriskjava.manager.internal.ManagerConnectionImpl.sendAction(ManagerConnectionImpl.java:936)
08:05:22,033 ERROR [stderr] (AJ DaemonPool-1.2) 	at org.asteriskjava.manager.internal.ManagerConnectionImpl.sendAction(ManagerConnectionImpl.java:896)
08:05:22,033 ERROR [stderr] (AJ DaemonPool-1.2) 	at org.asteriskjava.manager.DefaultManagerConnection.sendAction(DefaultManagerConnection.java:326)
08:05:22,033 ERROR [stderr] (AJ DaemonPool-1.2) 	at org.asteriskjava.live.internal.AsteriskServerImpl.sendAction(AsteriskServerImpl.java:981)
08:05:22,033 ERROR [stderr] (AJ DaemonPool-1.2) 	... 8 more
```

It has timeout in ChannelManager.addNewChannel(), because getVariable() function needs more time than 2s (which is the default value of ManagerConnectionImpl.defaultResponseTimeout).

Timeout happens in ChannelManager.java at line 223:
  channel.getVariable(Constants.VARIABLE_TRACE_ID);

Suggested **solution** (after debuging 2 days), ChannelManager.java at line 225
before:
``
                catch (NoSuchChannelException e)
``
after:
``
                catch (NoSuchChannelException | ManagerCommunicationException e)
``
				
This way, if a simple GetVarAction has timeout, it will try it again, and don't skip the rest of the addNewChannel(), because of an exception. (Or if it has no more time (SLEEP_TIME_BEFORE_GET_VAR interval was not enough to complete GetVar), it will skip it simple.)

I would happily upgrade to the latest version (instead of 2.0.0), but this bug blocks it.
Thank you very much! :)

**Comment by bsutton** [comment]
Can I suggest that you fork the project and push a pr.

We can then review the code.

Superficiality you suggested change sounds fine.

**Comment by endredy** [comment]
Hi,
the pull request was created, i hope it will help the code. :)
thanks!

**Comment by seanbright** [comment]
Closed by #303
